tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/VoteAPIHelper.java

```diff
@@ -0,0 +1,342 @@
+package org.unicode.cldr.web.api;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.unicode.cldr.test.CheckCLDR;
+import org.unicode.cldr.test.CheckCLDR.CheckStatus;
+import org.unicode.cldr.test.CheckCLDR.CheckStatus.Subtype;
+import org.unicode.cldr.test.CheckForCopy;
+import org.unicode.cldr.test.DisplayAndInputProcessor;
+import org.unicode.cldr.test.TestCache.TestResultBundle;
+import org.unicode.cldr.util.CLDRConfig;
+import org.unicode.cldr.util.CLDRFile;
+import org.unicode.cldr.util.CLDRInfo.CandidateInfo;
+import org.unicode.cldr.util.CLDRInfo.UserInfo;
+import org.unicode.cldr.util.CLDRLocale;
+import org.unicode.cldr.util.CldrUtility;
+import org.unicode.cldr.util.PathHeader;
+import org.unicode.cldr.util.PathHeader.PageId;
+import org.unicode.cldr.util.PathHeader.SurveyToolStatus;
+import org.unicode.cldr.util.SupplementalDataInfo;
+import org.unicode.cldr.web.BallotBox;
+import org.unicode.cldr.web.BallotBox.VoteNotAcceptedException;
+import org.unicode.cldr.web.CookieSession;
+import org.unicode.cldr.web.DataSection;
+import org.unicode.cldr.web.DataSection.DataRow;
+import org.unicode.cldr.web.DataSection.DataRow.CandidateItem;
+import org.unicode.cldr.web.STFactory;
+import org.unicode.cldr.web.SurveyException.ErrorCode;
+import org.unicode.cldr.web.SurveyLog;
+import org.unicode.cldr.web.SurveyMain;
+import org.unicode.cldr.web.UserRegistry;
+import org.unicode.cldr.web.XPathMatcher;
+import org.unicode.cldr.web.XPathTable;
+import org.unicode.cldr.web.api.VoteAPI.RowResponse;
+import org.unicode.cldr.web.api.VoteAPI.VoteResponse;
+
+import com.ibm.icu.dev.util.ElapsedTimer;
+import com.ibm.icu.util.Output;
+
+/**
+ * Note: The functions in this class needed to be separated from VoteAPI because of static init problems.
+ */
+public class VoteAPIHelper {
+    static final boolean DEBUG = false;
+    static Response handleGetRows(String loc, String xpath, String session) {
+        final SurveyMain sm = CookieSession.sm;
+        ElapsedTimer et = new ElapsedTimer();
+        final CLDRLocale locale = CLDRLocale.getInstance(loc);
+        // Verify session
+        final CookieSession mySession = Auth.getSession(session);
+        {
+            final Response r = Auth.assertSession(mySession);
+            if(r != null) return r;
+        }
```

Comment:
How about this (combining the ideas…)

```suggestion
        final CookieSession mySession = Auth.getSession(session);
        if (mySession == null) return Auth.invalidSessionResponse();
```